src/js/utils/AutoBackup.js

```diff
@@ -0,0 +1,137 @@
+        }
+    }
+
+    handleDisconnect(event) {
+        if (event.detail) {
+            gui_log(i18n.getMessage('serialPortClosedOk'));
+        } else {
+            gui_log(i18n.getMessage('serialPortClosedFail'));
+        }
+
+        serial.removeEventListener('receive', this.readSerialAdapter);
+        serial.removeEventListener('connect', this.handleConnect);
+        serial.removeEventListener('disconnect', this.handleDisconnect);
+    }
+
+    readSerialAdapter(info) {
+        const data = new Uint8Array(info.detail.buffer);
+
+        for (const charCode of data) {
+            const currentChar = String.fromCharCode(charCode);
+            this.outputHistory += currentChar;
+        }
+    }
+
+    onClose() {
+        serial.addEventListener('disconnect', this.handleDisconnect.bind(this), { once: true });
+        serial.disconnect();
+    }
+
+    async save() {
+        console.log('Saving backup');
+        const prefix = 'cli_backup';
+        const suffix = 'txt';
+        const text = this.outputHistory;
+        const filename = generateFilename(prefix, suffix);
+
+        FileSystem.pickSaveFile(filename, i18n.getMessage('fileSystemPickerFiles', { types: suffix.toUpperCase() }), `.${suffix}`)
+        .then((file) => {
+            console.log("Saving config to:", file.name);
+            FileSystem.writeFile(file, text);
+        })
+        .catch((error) => {
+            console.error("Error saving config:", error);
+        })
+        .finally(() => {
+            if (this.callback) {
+                this.callback();
+            }
+        });
+    }
+
+    async run() {
+        console.log('Running backup');
+
+        await this.activateCliMode();
+        await this.sendCommand("dump all");
+
+        setTimeout(async () => {
+            this.sendCommand("exit", this.onClose);
+            await this.save(this.outputHistory);
+        }, 1500);
+    }
+
+    async activateCliMode() {
+        return new Promise(resolve => {
+            const bufferOut = new ArrayBuffer(1);
+            const bufView = new Uint8Array(bufferOut);
+
+            bufView[0] = 0x23;
+
+            serial.send(bufferOut);
+
+            setTimeout(() => {
+                this.outputHistory = '';
+                resolve();
+            }, 500);
+        });
+    }
+
+    async sendSerial(line, callback) {
+        const bufferOut = new ArrayBuffer(line.length);
+        const bufView = new Uint8Array(bufferOut);
+
+        for (let cKey = 0; cKey < line.length; cKey++) {
+            bufView[cKey] = line.charCodeAt(cKey);
+        }
+
+        serial.send(bufferOut, callback);
+    }
+
+    async sendCommand(line, callback) {
+        this.sendSerial(`${line}\n`, callback);
+    }
+
+    execute(callback) {
+        this.callback = callback;
+
+        const port = PortHandler.portPicker.selectedPort;
+
+        if (port.startsWith('serial')) {
+            const baud = parseInt($('#flash_manual_baud_rate').val()) || 115200;
```

Comment:
This value is at the PortHandler too, not need to read it with jquery.

Reply:
`flash_manual_baud_rate` is not the same as `PortHandler.portPicker.selectedBauds`

![image](https://github.com/betaflight/betaflight-configurator/assets/8344830/79cc5388-c250-42ad-a31b-531a80fc6e7e)

Reply:
True, I didn't remember it. The backup must use this flash baud rate in place of the "connection" one?

Reply:
Updated to use the portPicker baud selection.